### PR TITLE
Validate distribution parameters

### LINKED ISSUE #1898: Logarithmic distribution: support for negative standard deviation?
This is more of a discussion placeholder. Currently all the log distributions convert standard deviation to positive number by using absolute value given in the priors file, eg.
```
LOG_PARAM LOGNORMAL 1 -10
```
would compute distribution as `exp(x * 10 + 1)==(e^1, e^11)` instead of  `exp(-x * 10 + 1)==(e^(-9), e^1)`, which yields quite different results. Shall we provide support for such small distributions too? 

Maybe this is not too much relevant for the reservoir domain but some other applications.

### PR DESCRIPTION
**Pull request by DanSava**
**Issue**
Resolves #1898 

**Approach**
Check that std is positive for NORMAL LOGNORMAL and TRUNCATED_NORMAL distributions 
Also check mean is positive for the LOGNORMAL distribution

## Pre review checklist

- [ ] Added appropriate release note label
- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Updated documentation
- [ ] Ensured new behaviour is tested

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: tests/unit_tests/c_wrappers/res/enkf/config/test_gen_kw_config.py [hunk 1]
```diff
@@ -199,3 +200,47 @@ def test_gen_kw_is_log_or_not(
...
+        GEN_KW KW_NAME template.txt kw.txt prior.txt
+        """
+        )
+        with open("config.ert", "w", encoding="utf-8") as fh:
+            fh.writelines(config)
+        with open("template.txt", "w", encoding="utf-8") as fh:
+            fh.writelines("MY_KEYWORD <MY_KEYWORD>")
+        with open("prior.txt", "w", encoding="utf-8") as fh:
+            if distribution == "TRUNCATED_NORMAL":
+                fh.writelines(f"MY_KEYWORD {distribution} {mean} {std} -1 1")
```
Comment (andreas-el) [inline tests/unit_tests/c_wrappers/res/enkf/config/test_gen_kw_config.py:235]: I guess testing of the hard limits for `TRUNCATED_NORMAL` is out of scope for this?
Maybe it does not make sense to test those?

**Review by andreas-el — APPROVED** [review]
Looks good to me. 👍 
Just had some questions because I'm curious.

File: src/ert/_c_wrappers/enkf/ensemble_config.py [hunk 2]
```diff
@@ -421,6 +421,34 @@ def alloc_keylist(self) -> StringList:
...
+                )
+
+        for prior in mode_config.get_priors():
+            if prior["function"] == "LOGNORMAL":
+                _check_non_negative_parameter("MEAN")
+                _check_non_negative_parameter("STD")
+            if prior["function"] == "NORMAL":
+                _check_non_negative_parameter("STD")
+            if prior["function"] == "TRUNCATED_NORMAL":
+                _check_non_negative_parameter("STD")
```
[the remark this answers is not in the document]
Comment (DanSava) [reply src/ert/_c_wrappers/enkf/ensemble_config.py:445]: Very good suggestion. I love it :heart:

File: tests/unit_tests/c_wrappers/res/enkf/config/test_gen_kw_config.py (hunk 1 above)
Comment (DanSava) [reply tests/unit_tests/c_wrappers/res/enkf/config/test_gen_kw_config.py:235]: Not really sure what are the hard limits of `TRUNCATED_NORMAL` to actually test them, but that is not the purpose of this test.

**Pull request merged**